pkg/operation/botanist/backupentry.go

```diff
@@ -90,3 +92,14 @@ func (b *Botanist) DeploySourceBackupEntry(ctx context.Context) error {
 	b.Shoot.Components.SourceBackupEntry.SetBucketName(bucketName)
 	return b.Shoot.Components.SourceBackupEntry.Deploy(ctx)
 }
+
+// DestroySourceBackupEntry destroys the source BackupEntry. It returns nil if the CopyEtcdBackupsDuringControlPlaneMigration feature gate
+// is disabled or the Seed backup is not enabled or the Shoot is in restore phase.
+func (b *Botanist) DestroySourceBackupEntry(ctx context.Context) error {
```

Comment:
Will this break similarly when the restoration flow failed after this `DestroySourceBackupEntry` function had been executed successfully before? Subsequent executions of the restoration flow will try to delete the `SourceBackupEntry` again and fail since it's already gone, or?

Reply:
It shouldn't since the `DeleteObject` function that this `Destroy` is actually invoking is ignoring both `NotFound` and `NoMatch` errors, so attempting to delete a missing backup entry should succeed.

Reply:
Yes, but the seed authorizer will deny the request since it won't know the object (it's not in the system, hence, no decision can be taken). Isn't this the same problem as with current `master`?

Reply:
Should we then perhaps treat `delete` as "always allowed" rather than "allowed" in #5066? If you are right, wouldn't we have similar issues with every resource we delete in the deletion flow, unless `delete` is "always allowed"?

Reply:
@rfranzke if you mean whether the CP migration can still fail with `no relationship found between seed 'foo' and this object` (this source BackipEntry) - I guess yes, I have to test it out. But let's leave the SeedAuthorizer improvements for PR https://github.com/gardener/gardener/pull/5066.

Reply:
I think the only resource we delete in the garden cluster is the `ShootState` at the very end of the flow (https://github.com/gardener/gardener/blob/3678f65531c172e56fffc65d7c27d82559203513/pkg/gardenlet/controller/shoot/shoot_control_delete.go#L615-L619), but yes, probably the same problem does theoretically apply here as well. 

> Should we then perhaps treat delete as "always allowed" rather than "allowed" in #5066?

This is dangerous because it would allow all gardenlets to delete arbitrary `BackupEntry`s without any relationship checks. Probably the seed authorizer can be generally improved to allow `DELETE` verbs in case the respective node is not found in the graph. We should tackle this probably separate from this PR since it's a corner case that can already happen with the `ShootState` as well.

Reply:
OK, we clarified why #5066 worked with a local dev setup (my question above), basically there is a flow in the local test setup. As @ialidzhikov suggested, let's leave the discussion about seed authorizer improvements to that PR.